qrmi/src/pasqal/cloud.rs

```diff
@@ -0,0 +1,659 @@
+            Ok(()) => Ok(()),
+            Err(v) => Err(v.into()),
+        }
+    }
+
+    /// Python binding of QRMI task_start() function.
+    #[pyo3(name = "task_start")]
+    fn pyfunc_task_start(&mut self, payload: Payload) -> PyResult<String> {
+        match self.task_start(payload) {
+            Ok(v) => Ok(v),
+            Err(v) => Err(v.into()),
+        }
+    }
+
+    /// Python binding of QRMI task_stop() function.
+    #[pyo3(name = "task_stop")]
+    fn pyfunc_task_stop(&mut self, task_id: &str) -> PyResult<()> {
+        match self.task_stop(task_id) {
+            Ok(()) => Ok(()),
+            Err(v) => Err(v.into()),
+        }
+    }
+
+    /// Python binding of QRMI task_status() function.
+    #[pyo3(name = "task_status")]
+    fn pyfunc_task_status(&mut self, task_id: &str) -> PyResult<TaskStatus> {
+        match self.task_status(task_id) {
+            Ok(v) => Ok(v),
+            Err(v) => Err(v.into()),
+        }
+    }
+
+    /// Python binding of QRMI task_result() function.
+    #[pyo3(name = "task_result")]
+    fn pyfunc_task_result(&mut self, task_id: &str) -> PyResult<TaskResult> {
+        match self.task_result(task_id) {
+            Ok(v) => Ok(v),
+            Err(v) => Err(v.into()),
+        }
+    }
+
+    /// Python binding of QRMI target() function.
+    #[pyo3(name = "target")]
+    fn pyfunc_target(&mut self) -> PyResult<Target> {
+        match self.target() {
+            Ok(v) => Ok(v),
+            Err(v) => Err(v.into()),
+        }
+    }
+
+    /// Python binding of QRMI metadata() function.
+    #[pyo3(name = "metadata")]
+    fn pyfunc_metadata(&mut self) -> PyResult<HashMap<String, String>> {
+        let metadata: HashMap<String, String> = HashMap::new();
+        Ok(metadata)
+    }
+}
+
+impl Default for PasqalCloud {
+    fn default() -> Self {
+        Self::new("")
+    }
+}
+/// QuantumResource Trait implementation for PasqalCloud
+impl PasqalCloud {
+    /// Wrapper of async call for QRMI is_accessible() function.
+    #[tokio::main]
+    async fn _is_accessible(&mut self) -> bool {
+        let fresnel = DeviceType::Fresnel.to_string();
+        if self.backend_name != fresnel {
+            let err = format!(
+                "Device {} is invalid. Only {} device can receive jobs.",
+                self.backend_name, fresnel,
+            );
+            panic!("{}", err);
+        };
+        match self.api_client.get_device(DeviceType::Fresnel).await {
+            Ok(device) => device.data.status == "UP",
+            Err(_err) => false,
+        }
+    }
+
+    /// Wrapper of async call for QRMI task_start() function.
+    #[tokio::main]
+    async fn _task_start(&mut self, payload: Payload) -> Result<String> {
+        if let Payload::PasqalCloud { sequence, job_runs } = payload {
+            // TODO: Make configurable (get emulator from qrmi)
+            match self
+                .api_client
+                .create_batch(sequence, job_runs, DeviceType::EmuFree)
```

Comment:
here we hard-code the target for now. 
I want to keep this during the Spank development, after which we can fix it.

One issue we have is that we separate the device specifications to target from from the backend to target. But this only affects emulators, so I'm not sure if we want to do that here or not yet.